devel/shards/Portfile

```diff
@@ -16,9 +16,20 @@ long_description    Shards is a dependency manager for the Crystal Programming L
 
 depends_lib         port:crystal
 
-checksums           rmd160  ebefd07b9c067315b9804d8ebc9a657a24d1d70d \
-                    sha256  12b6483852b8cd487e4639f3ccae7fb39f26602406038ef292ed95713b561425 \
-                    size    41277
+set molinillo_ver   0.1.0
+set molinillo_dist  crystal-molinillo-${molinillo_ver}
+
+master_sites-append https://github.com/crystal-lang/crystal-molinillo/archive/:molinillo
+distfiles-append    v${molinillo_ver}${extract.suffix}:molinillo
```

Comment:
If you're going to use the archive tarball for crystal-molinillo (and that's the one we want people to use now), you should also use the archive tarball for shards (rather than the tags tarball that you're using now because it is the default of the github portgroup). You can get this for the main download by using `github.tarball_from archive`. You will need to delete the main distfile and get download to download the new one so that you can compute the new checksums and put them into the portfile.

You've specified a tag (`molinillo`) for this distfile and its master_site, which is good because it will ensure that MacPorts will only attempt to download this distfile from this master_site (and our mirrors). But you haven't specified a tag for the main distfile or its master_site (the ones the github portgroup set for you automatically), which means that MacPorts could attempt to download the main distfile from molinillo's master_site, *which may unexpectedly succeed* and result in a checksum mismatch. Unfortunately the github 1.0 portgroup was not designed to assist you in downloading more than one distfile from more than one project, so a bit of extra manual work is needed in the portfile. There are a few examples you could look at.

When we have a choice—and with github we do because they let us specify any distname we want—we do like to put both the project name and the version number into the distname. That's especially valuable here, since you're downloading the distfiles of two different projects and it would be nice to know by looking at the filename which is which. Look at the URL MacPorts constructs for your main distfile once you set `github.tarball_from archive` and mimic that structure for any additional distfiles.

Reply:
Thanks for the detailed feedback. I've implemented the changes you mentioned. Could you take another look at it?